system/nxcamera/nxcamera.c

```diff
@@ -0,0 +1,889 @@
+ *   that device.
+ *
+ * Input:
+ *   pcam    Pointer to the initialized Looper context
+ *   width      Capture frame width
+ *   height     Capture frame height
+ *   framerate  Capture frame rate
+ *   format     Capture frame pixel format
+ *
+ * Returns:
+ *   OK         Video is being looped
+ *   -EBUSY     Capture device is busy
+ *   -ENOSYS    No supported video format found
+ *   -ENODEV    No video capture or framebuffer device suitable
+ *
+ ****************************************************************************/
+
+int nxcamera_stream(FAR struct nxcamera_s *pcam,
+                    uint16_t width, uint16_t height,
+                    uint32_t framerate, uint32_t format)
+{
+  struct mq_attr              attr;
+  struct sched_param          sparam;
+  pthread_attr_t              tattr;
+  FAR void                    *value;
+  int                         ret;
+  int                         i;
+  struct v4l2_buffer          buf;
+  struct v4l2_requestbuffers  req;
+  struct v4l2_streamparm      parm;
+
+  DEBUGASSERT(pcam != NULL);
+
+  if (pcam->loopstate != NXCAMERA_STATE_IDLE)
+    {
+      return -EBUSY;
+    }
+
+  vinfo("==============================\n");
+  vinfo("streaming video\n");
+  vinfo("==============================\n");
+
+  /* Try to open the device */
+
+  ret = nxcamera_opendevice(pcam);
+  if (ret < 0)
+    {
+      /* Error opening the device */
+
+      verr("ERROR: nxcamera_opendevice failed: %d\n", ret);
+      return ret;
+    }
+
+  /* VIDIOC_S_FMT set format */
+
+  pcam->fmt.type                = V4L2_BUF_TYPE_VIDEO_CAPTURE;
+  pcam->fmt.fmt.pix.width       = width;
+  pcam->fmt.fmt.pix.height      = height;
+  pcam->fmt.fmt.pix.field       = V4L2_FIELD_ANY;
+  pcam->fmt.fmt.pix.pixelformat = format;
+
+  ret = ioctl(pcam->capture_fd, VIDIOC_S_FMT, (uintptr_t)&pcam->fmt);
+  if (ret < 0)
+    {
+      ret = -errno;
+      verr("VIDIOC_S_FMT failed: %d\n", ret);
+      return ret;
+    }
+
+  memset(&parm, 0, sizeof(parm));
+  parm.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
+  parm.parm.capture.timeperframe.denominator = framerate;
+  parm.parm.capture.timeperframe.numerator = 1;
+  ret = ioctl(pcam->capture_fd, VIDIOC_S_PARM, (uintptr_t)&parm);
+  if (ret < 0)
+    {
+      ret = -errno;
+      verr("VIDIOC_S_PARM failed: %d\n", ret);
+      return ret;
+    }
+
+  /* VIDIOC_REQBUFS initiate user pointer I/O */
+
+  req.type   = V4L2_BUF_TYPE_VIDEO_CAPTURE;
+  req.memory = V4L2_MEMORY_MMAP;
+  req.count  = CONFIG_VIDEO_REQBUFS_COUNT_MAX;
+
+  ret = ioctl(pcam->capture_fd, VIDIOC_REQBUFS, (uintptr_t)&req);
+  if (ret < 0)
+    {
+      ret = -errno;
+      verr("VIDIOC_REQBUFS failed: %d\n", ret);
+      return ret;
+    }
+
+  if (req.count < 2)
+    {
+      verr("VIDIOC_REQBUFS failed: not enough buffers\n");
+      return -ENOMEM;
+    }
+
+  pcam->nbuffers = req.count;
+  pcam->bufs = calloc(req.count, sizeof(*pcam->bufs));
+  pcam->buf_sizes = calloc(req.count, sizeof(*pcam->buf_sizes));
```

Comment:
Just to keep same style with other places
```suggestion
  pcam->nbuffers  = req.count;
  pcam->bufs      = calloc(req.count, sizeof(*pcam->bufs));
  pcam->buf_sizes = calloc(req.count, sizeof(*pcam->buf_sizes));
```